src/main/java/me/jellysquid/mods/sodium/mixin/features/mipmaps/MixinSpriteContents.java

```diff
@@ -0,0 +1,138 @@
+package me.jellysquid.mods.sodium.mixin.features.mipmaps;
+
+import net.minecraft.client.texture.NativeImage;
+import net.minecraft.client.texture.SpriteContents;
+import net.minecraft.util.Identifier;
+import org.objectweb.asm.Opcodes;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Mutable;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+/**
+ * This Mixin is ported from Iris at <a href="https://github.com/IrisShaders/Iris/blob/41095ac23ea0add664afd1b85c414d1f1ed94066/src/main/java/net/coderbot/iris/mixin/bettermipmaps/MixinTextureAtlasSprite.java">MixinTextureAtlasSprite</a>.
+ */
+@Mixin(SpriteContents.class)
+public class MixinSpriteContents {
+    @Mutable
+    @Shadow
+    @Final
+    private NativeImage image;
+    // Generate some color tables for gamma correction.
+    private static final float[] SRGB_TO_LINEAR = new float[256];
+
+    static {
+        for (int i = 0; i < 256; i++) {
+            SRGB_TO_LINEAR[i] = (float) Math.pow(i / 255.0, 2.2);
+        }
+    }
+
+    // While Fabric allows us to @Inject into the constructor here, that's just a specific detail of FabricMC's mixin
+    // fork. Upstream Mixin doesn't allow arbitrary @Inject usage in constructor. However, we can use @ModifyVariable
+    // just fine, in a way that hopefully doesn't conflict with other mods.
+    //
+    // By doing this, we can work with upstream Mixin as well, as is used on Forge. While we don't officially
+    // support Forge, since this works well on Fabric too, it's fine to ensure that the diff between Fabric and Forge
+    // can remain minimal. Being less dependent on specific details of Fabric is good, since it means we can be more
+    // cross-platform.
+    @Redirect(method = "<init>", at = @At(value = "FIELD", target = "Lnet/minecraft/client/texture/SpriteContents;image:Lnet/minecraft/client/texture/NativeImage;", opcode = Opcodes.PUTFIELD))
+    private void iris$beforeGenerateMipLevels(SpriteContents instance, NativeImage nativeImage, Identifier identifier) {
+        // We're injecting after the "info" field has been set, so this is safe even though we're in a constructor.
+        if (identifier.getPath().contains("leaves")) {
```

Comment:
We should provide a resource override rather than hardcoding this so that mods have an example of what to do.